Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7949.cs

```diff
@@ -0,0 +1,135 @@
+	public class Issue7949 : TestContentPage
+	{
+		readonly Aspect[] _aspectSettings = { Aspect.AspectFit, Aspect.AspectFill, Aspect.Fill };
+		const string CycleButtonAutomationId = "CycleAspectsThingaMejiggy";
+
+		protected override void Init()
+		{
+			var grid = new Grid
+			{
+				BackgroundColor = Color.Yellow
+			};
+
+			grid.RowDefinitions.Add(new RowDefinition
+			{
+				Height = GridLength.Star
+			});
+
+			grid.RowDefinitions.Add(new RowDefinition
+			{
+				Height = GridLength.Star
+			});
+
+			grid.ColumnDefinitions.Add(new ColumnDefinition
+			{
+				Width = GridLength.Star
+			});
+
+			grid.ColumnDefinitions.Add(new ColumnDefinition
+			{
+				Width = GridLength.Star
+			});
+
+			var image0 = new Image
+			{
+				Source = "photo.jpg",
+				Aspect = Aspect.AspectFill,
+				BackgroundColor = Color.Red,
+				HorizontalOptions = LayoutOptions.FillAndExpand,
+				VerticalOptions = LayoutOptions.FillAndExpand
+			};
+
+			var image1 = new Image
+			{
+				Source = "photo.jpg",
+				Aspect = Aspect.AspectFit,
+				BackgroundColor = Color.Green,
+				HorizontalOptions = LayoutOptions.FillAndExpand,
+				VerticalOptions = LayoutOptions.FillAndExpand
+			};
+
+			var image2 = new Image
+			{
+				Source = "photo.jpg",
+				Aspect = Aspect.Fill,
+				BackgroundColor = Color.Blue,
+				HorizontalOptions = LayoutOptions.FillAndExpand,
+				VerticalOptions = LayoutOptions.FillAndExpand
+			};
+
+			var stack = new StackLayout();
+
+			var image3 = new Image
+			{
+				Source = "photo.jpg",
+				Aspect = Aspect.AspectFit,
+				BackgroundColor = Color.Blue,
+				HorizontalOptions = LayoutOptions.FillAndExpand,
+				VerticalOptions = LayoutOptions.FillAndExpand
+			};
+
+			var button = new Button
+			{
+				AutomationId = CycleButtonAutomationId,
+				Text = "Cycle"
+			};
+
+			button.Clicked += (o, a) =>
+			{
+				var i = _aspectSettings.IndexOf(image3.Aspect) + 1;
+
+				if (i == _aspectSettings.Length)
+					i = 0;
+
+				image3.Aspect = _aspectSettings[i];
+			};
+
+			stack.Children.Add(image3);
+			stack.Children.Add(button);
+
+			grid.Children.Add(image0, 0, 0);
+			grid.Children.Add(image1, 1, 0);
+			grid.Children.Add(image2, 0, 1);
+			grid.Children.Add(stack, 1, 1);
+
+			Content = grid;
+		}
+
+#if UITEST
+		[Test]
+		[UiTest (typeof(Image))]
+		public async Task Issue852TestsEntriesClickable()
+		{
+			RunningApp.WaitForElement (CycleButtonAutomationId);
+			RunningApp.Screenshot("All aspects should show properly, lower right should show AspectFit");
+
+			RunningApp.Tap(CycleButtonAutomationId);
+			await Task.Delay(1000);
+			RunningApp.Screenshot("Lower right should show AspectFill");
+
+			RunningApp.Tap(CycleButtonAutomationId);
+			await Task.Delay(1000);
+			RunningApp.Screenshot("Lower right should show Fill");
+
+			RunningApp.Tap(CycleButtonAutomationId);
+			await Task.Delay(1000);
+			RunningApp.Screenshot("Lower right should show AspectFit");
```

Comment:
You can validate some properties on MacOS to previne regression.

Something like that:

```C#

#if __MACOS__
var nsImageScale = RunningApp.Query(c => c.Marked(YOUR_IMAGE_AUTOMATIONID).Invoke("ImageScaling ").Value<int>());

Assert.AreEqual(1, nsImageScale );
#endIf
```

[Documentation](https://docs.microsoft.com/en-us/dotnet/api/xamarin.uitest.queries.appquery.invoke?view=xamarin-uitest-sdk)

[My pr with similar test](https://github.com/xamarin/Xamarin.Forms/pull/7966/files#diff-c8584e3f1ae3f5ab5220d09f4873ea1aR69)

[StackOverflow question](https://stackoverflow.com/questions/45878350/xamarin-ui-test-take-controls-background-color-ios)

[Xamarin Forum question](https://forums.xamarin.com/discussion/61782/check-button-color-with-xamarin-uitest)